### DISCUSSION TITLE
Is there additional guide for font stretch?

### DISCUSSION THREAD

**Discussion by aca** [wezterm/wezterm#1644] [Q&A] [answered]
Hi,
I'm not sure whether this is an bug or just my misunderstanding.
I expect font stretch option to show font rendered like this https://developer.mozilla.org/en-US/docs/Web/CSS/font-stretch

But it simply doesn't work at all. I'm currently trying with nerdfont.
Is there additional requirements(settings/fonts) to make it work?

```
wezterm 20220213-203140-4a1c4b55
```

**Comment by wez** [comment, marked as answer]
are you setting the `stretch` parameter, something like this?

```lua
local wezterm = require 'wezterm';

return {
  font = wezterm.font('Iosevka Term', {stretch="Expanded", weight="Regular"}),
}
```

`stretch` is an instruction to find a variation on the requested font family that has the requested stretch attribute.  If you don't have a version of that font with the desired stretch then the closest alternative font variant will be used.  Stretch cannot be synthesized, so if there are no alternative stretched versions of the font installed, then setting stretch will have no effect.

I would recommend avoiding using patched nerd fonts, as the patched fonts are often a bit buggy and can cause conflicts with the real versions of the fonts.  wezterm's nightly builds ship with a bundled copy of the nerd font symbols as a fallback for any font, so you can just use the normal version of the font you want to use and you'll still get your nerd font icons.

**Reply by aca** [reply]
I was using source code pro (which has wide width) in alacritty with font width condensed with this.
```
font:
  offset:
    x: -3
    y: 0
```
It was quite lovely feature. I can't find fonts that have stretch attribute, Maybe can we add this in wezterm?

**Reply by rahil627** [reply]
hmmm, i've got the same problem on a customized source code pro: [office code pro](https://github.com/nathco/Office-Code-Pro)...

i don't recall seeing alternate sets of fonts based on stretch/width... just font thickness:

https://fonts.google.com/specimen/Source+Code+Pro

.. but i'm no font expert..

(been living on monofoki/mononoki font...)

**Reply by rahil627** [reply]
oh wait, it works, never-mind!! :D OMG. wezterm's detail on appearance, is incredible!!

**Reply by rahil627** [reply]
oh wait, it was a fluke, nvm 😓

**Reply by rahil627** [reply]
honestly.. i feel like vs-code is doing some magic to make fonts more condensed. Even Consolas is much more condensed in vs-code than wezterm.
